grafana/grafana
# Logs: Allow collapsing the logs volume histogram

Fixes https://github.com/grafana/grafana/issues/50265

allows the user to collapse the logs-volume-histogram widget. details:
- the histogram is enabled by default
- if you close it, we remember it in the browser's local-storage
- when it is collapsed, the histogram-query is not sent to the database (in other words: when histogram is open, and you run a logs-query, we send two queries to the database, one for logs, one for histogram. when histogram is collapsed, and you run a logs-query, we only send a single query to the database, for logs)

how to test:
(this test-scanario involves Loki, but it works with Elastic too)
1. `make devenv sources=loki`
2. go to the explore-page
3. open the browser-dev-tools, we will be watching the ajax-requests sent.
4. run the query `{source="data"}`
5. you should see logs, and you should see the logs-histogram
6. in the dev-tools you should see 2 new ajax requests named `query`, if you open them and check the query-data, one sends `expr: "{source="data"}"` , and the other sends `expr: "sum by ....`
7. now collapse the logs-volume section witht the `>` icon at the top-left corner of the widget
8. run the query from [4] again
9. check the ajax-requests. verify that only one request was sent, the one with the `expr:"{source...`.
10. now reload the webpage
11. verify that the logs-volume section remains collapsed
12. run the query from [4] again
13. clear the list of ajax-queries in the devtools
14. open the logs-volume-histogram section
15. verify that an ajax request was sent at this point with the `expr: "sum by..." content
16. verify that the logs-volume-histogram appears
